src/buffer.cc

```diff
@@ -575,9 +575,9 @@ BufferCoord Buffer::advance(BufferCoord coord, ByteCount count) const
         count += coord.column;
         while (count < 0)
         {
-            count += m_lines[--line].length();
-            if (line < 0)
+            if (--line < 0)
                 return {0, 0};
+            count += m_lines[line].length();
```

Comment:
nice find - `m_lines[-1]` is undefined behavior, so we should put the bounds check first.
The symmetry with the above case (with ++ instead of --) is deceptive - that one is fine but this one isn't.

Do you have a command that reproduces the crash?

Reply:
The command that caused this crash was 
```
./kak -f 'o/<c-n><up><c-n>' </dev/null
```
I was going to post it, but I completely forgot to do that.

Reply:
That only works with -f btw.